### PR TITLE
[Pick #3424] Relax ImageSet validation (r1.34)

### PR DESCRIPTION
**Pull request by tmjd**
## Description

Only block progress when an ImageSet exists for the same variant but not one with the correct version.

These changes help ensure when a cluster using Calico is upgrade to Enterprise that an ImageSet that had been created for Calico will not block the upgrade to Enterprise.
This helps decouple the management of a Calico install that was managed by a platform where the platform added an ImageSet but the cluster had been upgraded to Enterprise, it ensures the addition of the ImageSet for Calico won't impact the already functioning Enterprise cluster that is being managed separately.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Review by rene-dekker — APPROVED** [review]
lgtm

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**
